CMakeLists.txt

```diff
@@ -0,0 +1,180 @@
+    #
+    # This will add to the project using `add_subdirectory`.
+    #
+    # OpenAssetIO must therefore be built, meaning its `find_package`
+    # calls for dependencies must work.
+
+    add_subdirectory(OpenAssetIO)
+else ()
+    # Use OpenAssetIO as an external CMake package
+    #
+    # The `OpenAssetIOConfig.cmake` file must be discoverable. E.g.
+    # add the install directory (`dist`) to CMAKE_PREFIX_PATH.
+
+    find_package(OpenAssetIO REQUIRED)
+endif ()
+
+# For Windows tests to work we must add the dlls to the PATH
+if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+    file(TO_NATIVE_PATH "${OpenAssetIO_BINARY_DIR}" OpenAssetIO_BINARY_DIR_NATIVE)
+endif ()
+
+#-----------------------------------------------------------------------
+# C++ tests.
+
+add_executable(test.cpp.core src/test.core.cpp)
+add_test(cpp.core test.cpp.core)
+target_link_libraries(test.cpp.core PRIVATE OpenAssetIO::openassetio-core)
+
+
+# GCC's C++11 ABI toggle must match OpenAssetIO's settings.
+if (DEFINED OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI)
+    if (OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI)
+        target_compile_definitions(test.cpp.core PRIVATE _GLIBCXX_USE_CXX11_ABI=1)
+    else ()
+        target_compile_definitions(test.cpp.core PRIVATE _GLIBCXX_USE_CXX11_ABI=0)
+    endif ()
+endif ()
+
+if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+    set_tests_properties(cpp.core PROPERTIES ENVIRONMENT "PATH=${OpenAssetIO_BINARY_DIR_NATIVE}")
+endif ()
+
+#-----------------------------------------------------------------------
+# C tests.
+
+if (OPENASSETIOTEST_ENABLE_C)
+    add_executable(test.c.core src/test.core.c)
+    add_test(c.core test.c.core)
+    target_link_libraries(test.c.core PRIVATE OpenAssetIO::openassetio-core-c)
+    if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+        set_tests_properties(c.core PROPERTIES ENVIRONMENT "PATH=${OpenAssetIO_BINARY_DIR_NATIVE}")
+    endif ()
+endif ()
+
+#-----------------------------------------------------------------------
+# Python tests.
+
+if (OPENASSETIOTEST_ENABLE_PYTHON)
+
+    if (OPENASSETIOTEST_ENABLE_SUBPROJECT)
+        message(
+            FATAL_ERROR
+            "Testing OpenAssetIO as a subproject when Python is enabled is currently unsupported."
+            " The binary Python module must be built and made available alongside the Python"
+            " sources for tests to work."
+        )
+    endif ()
+
+    find_package(Python COMPONENTS Interpreter Development.Embed REQUIRED)
+
+    if (NOT "${Python_VERSION}" STREQUAL "${OpenAssetIO_Python_VERSION}")
+        message(
+            FATAL_ERROR
+            "Found Python '${Python_VERSION}' but OpenAssetIO was built against"
+            " '${OpenAssetIO_Python_VERSION}'"
+        )
+    endif ()
+
+    #-----------------------------------------------------------------------
+    # Python sources tests.
+
+    # Python sources test.
+    add_test(NAME python COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/src/test.py)
```

Comment:
In other Python tests we've written, we've added

-v - verbose, only shown in CTest failed mode, but sometimes useful in debugging
-B - don't write .pycache files (to know you're always having to reinterpret script on re-runs of the tests
-s - ignore the user site-directory (to avoid unexpected things leaking in)

so just adding them here for consideration.

Reply:
I found that `-v` was an insane amount of logging!  So I went with just `-Bs` in 8c253c09

Reply:
But you only see that on a failure?

Reply:
I guess that's true (I always run with `--extra-verbose` locally), but even then I'd worry about that much output obfuscating the actual error.  

I recall when I discovered the need to set `PYTHONHOME` that the error spat out by Python was quite informative and verbose.

Do you really think `-v` is more useful than noise?

Reply:
It _has_ been useful. But if you're more comfortable without it, I won't press.

We're mostly debugging when thirdparty libraries aren't behaving, as opposed to our own code in the equivalent tests, hence more logging is useful.

Reply:
OK, happy to defer to experience - added `-v` in 1d39c428